### PR TITLE
integrate UserHelper with the rest

### PR DESCRIPTION
**Pull request by koko-ng**
Just some code to integrate the authentication process into our current app.

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/blindly/BlindlyApplication.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import android.app.Application
+import android.content.res.Configuration
+import dagger.hilt.android.HiltAndroidApp
+
+// Custom app to let hilt hook into
+@HiltAndroidApp
+class BlindlyApplication : Application() {
+
+}
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/BlindlyApplication.kt:11]: Why is this empty class relevant ?

File: app/src/main/java/ch/epfl/sdp/blindly/utils/UserHelper.kt [hunk 2]
```diff
@@ -6,109 +6,123 @@ import android.content.Intent
...
+
+        return AuthUI.getInstance()
+            .createSignInIntentBuilder()
+            .setAvailableProviders(providers)
+            //.setLogo(R.drawable.my_great_logo) // Set logo drawable
+            .setTheme(R.style.Theme_Blindly) // Set theme
+            /*.setTosAndPrivacyPolicyUrls(
+    "https://example.com/terms.html",
+    "https://example.com/privacy.html")*/
+            .build();
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/utils/UserHelper.kt:58]: Might be a good idea to clean up these comments and keep only what is useful (or explain why it is in comments for now).

**Review by Xelowak — CHANGES_REQUESTED** [review]
Might be useful to explain why all `@RunWith(AndroidJUnit4::class)` have been replaced by  `@HiltAndroidTest`, and how it works.  
Will we have to do that for each new test class that we will write in the future ?

In general, I think it is quite difficult to understand everything that has been done by reading your code. You should consider commenting it more since others (including me) don't know anything about these libraries and annotations that don't seem to be really instinctive (like Hilt).

File: app/src/main/java/ch/epfl/sdp/blindly/BlindlyApplication.kt (hunk 1 above)
Comment (koko-ng) [reply app/src/main/java/ch/epfl/sdp/blindly/BlindlyApplication.kt:11]: Just read the comment above ;)

File: app/src/main/java/ch/epfl/sdp/blindly/utils/UserHelper.kt (hunk 2 above)
Comment (koko-ng) [reply app/src/main/java/ch/epfl/sdp/blindly/utils/UserHelper.kt:58]: Well for the comments it is just have these codes available when we will need them, I think adding login by phone could  interset us. Then we will probably add ToS and policy one day.

I admit that I could already add the logo now.

**Comment by koko-ng** [comment]
> Might be useful to explain why all `@RunWith(AndroidJUnit4::class)` have been replaced by `@HiltAndroidTest`, and how it works.
> Will we have to do that for each new test class that we will write in the future ?

Well I don't know what is more intuitive between RunWith and HiltAndroidTest, the first one literally tells that it runs a with Android JUnit while the second tells that it's an android test with hilt. I mean I don't think it brings a lot of difference, and commenting everything with "hook hilt into this test" looks quite redundant to me...

File: app/src/main/java/ch/epfl/sdp/blindly/BlindlyApplication.kt (hunk 1 above)
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/BlindlyApplication.kt:11]: Is this to inject manual dependencies?
I'm not sure I understand how this works... Couldn't we do without this empty class?

**Review by cmsigrist — APPROVED** [review]

**Comment by Xelowak** [comment]
> > Might be useful to explain why all `@RunWith(AndroidJUnit4::class)` have been replaced by `@HiltAndroidTest`, and how it works.
> > Will we have to do that for each new test class that we will write in the future ?
> 
> Well I don't know what is more intuitive between RunWith and HiltAndroidTest, the first one literally tells that it runs a with Android JUnit while the second tells that it's an android test with hilt. I mean I don't think it brings a lot of difference, and commenting everything with "hook hilt into this test" looks quite redundant to me...

I didn't mean that you had to comment all annotations. However, you could tell us the changes you made (and why) in the description of the pull resquest.

**Review by Xelowak — APPROVED** [review]
We talked about these changes in our meeting, it's ok for me.

**Pull request merged**